Assets/Scripts/DeviceCommunicator.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using System.Text;
+using UnityEngine;
+using Zenject;
+
+public class DeviceCommunicator : MonoBehaviour {
+
+	[Inject]
+	CodeExecutor codeExecutor;
+
+	public void Listen(string protobufInstructionsString) 
+	{
+		byte[] protobufInstructions = Encoding.Default.GetBytes (protobufInstructionsString);
+		Code code = Code.Parser.ParseFrom (protobufInstructions);
+
+		codeExecutor.Run (code);
+	}
+
+	[DllImport ("__Internal")]
+	private extern static void endLevel(LevelCompleteStatus status);
```

Comment:
is this a lowercase `e` because it will be the function we implement in a `.mm` file in iOS?

Reply:
Taken from https://blogs.unity3d.com/2015/07/02/il2cpp-internals-pinvoke-wrappers/ on marshalling:
```
[DllImport("__Internal")]
private extern static int Increment(int value);
```

Reply:
Ah I was talking about the `e` on `endLevel`

Reply:
Ah. Uhm, I don't know. I think so, I've found examples with both so I assume it's a C++ reference rather than a Unity one.